leaderboardsgg/leaderboard-site
# Create user profile header

Implements a user profile header, which shows up when visiting a user page.

## What

* Re-adds the Nuxt 2 user profile header components, which have taken place in this repository in the past. Slight changes to some of them are done. Currently, only the username is dynamic.
* The user page (`pages/user/[id].vue`) imports the `ProfileHeader` component and embeds it.
* Icons of the 4 current social media buttons are added to `public/icons/social`.

## Link to Issue

Closes https://github.com/leaderboardsgg/leaderboard-site/issues/542

## Acceptance

### Steps for testing

Set up the frontend and backend as usual. Create a user and navigate to `/user/{id}`. The page should contain a user profile header, where the username is dynamic.

### Images

| Mobile design | Desktop design |
| --- | --- |
| <img width="460" alt="Mobile design" src="https://user-images.githubusercontent.com/78196474/222232724-05e8afae-2e04-4154-8263-17af7c8c55e9.png"> | <img width="822" alt="Desktop design" src="https://user-images.githubusercontent.com/78196474/222233903-41bd8e95-6215-4239-94e6-f634b91e117f.png"> |
